conn.go

```diff
@@ -75,7 +76,8 @@ type Conn struct {
 
 	handshakeMessageHandler        handshakeMessageHandler
 	flightHandler                  flightHandler
-	handshakeCompletedSignal       chan bool
+	handshakeCtx                   context.Context
```

Comment:
Mind choosing a name that is a little more descriptive and putting these members together? On my first read through I was confused if `handshakeCtx` and `handshakeDoneSignal` were related.

Maybe `handshakeCtx` and `handshakeCtxCancel` (or anything like that) just want to make it clear to the reader!